libsolidity/formal/Predicate.cpp

```diff
@@ -575,9 +595,9 @@ map<string, optional<string>> Predicate::readTxVars(smtutil::Expression const& _
 		{"block.number", TypeProvider::uint256()},
 		{"block.timestamp", TypeProvider::uint256()},
 		{"blockhash", TypeProvider::array(DataLocation::Memory, TypeProvider::uint256())},
-		{"msg.data", TypeProvider::bytesMemory()},
+		{"msg.data", TypeProvider::array(DataLocation::Memory, TypeProvider::fixedBytes(1))},
```

Comment:
It is `calldata` in 
https://github.com/ethereum/solidity/blob/9d6eaa7a46425301ddb961a369b04f211d53691c/libsolidity/ast/Types.cpp#L3963

But not sure if that would be a problem here.

Maybe the `fixedBytes` argument can also be avoided?

Reply:
Would that make the element type a `fixedBytes(1)` implicitly?

Reply:
We need that to report the type properly as hex.

Reply:
Yes, `fixedBytes(1)` is implicit.

Reply:
Ok, changed and squashed.